### DISCUSSION TITLE
adding/removing address books within a preset, adding individual address books manually

### DISCUSSION THREAD

**Discussion by hpkunz** [blind-coder/rcmcarddav#364] [General]
Hi,

first, thanks for this plugin. I am new to this, so please forgive me, if my question is stupid.

I am using roundcube 1.5 and sogo 4.3.2 as caldav server.

I configured a preset which loads all users address books the first time the user logs in. this works very nicely (and takes some time as some calendars are large, 500 entries).

my question is, what happens if the user adds an address book or removes one. when I tried it, new address books never appear and deleted address books were not removed (after loging our and in again). this is in accordance with the documentation and I also checked the code and I see, that it is apparently supposed to be like this.

To be honest, this makes the plugin pretty useless to me, which is a shame, as I really like to be able to use our caldav address books in roundcube.

what is the reason, that this is implemented like this?

=======

is there another way to configure a users address books, such that the addition and removal of address books is reflected in roundcube?

I tried to add the the address books manually in the caldav settings section. when I add an address book, *all* address books of the user are discovered and added, which is nice. here the user can remove an address book manually, which was deleted on the server, which is not optimal, but could live with that. 

however, when I try to add an additional address book manually (which was previously created on the caldav server), the plugin adds all the address books again (again doing a discovery, even if I provide a specific URL). so I end up with quite a few duplicate address books. of course I can remove all duplicates manually, but this seems meaningless to me.

=======

am I missing something? 

again, it is not my intention the rant here, on the contrary. I would also be glad to help, if that should be necessary.

Best,
Hp

**Comment by mstilkerich** [comment]
Well, the short answer is because nobody has implemented it yet. Rediscovery of addressbooks is planned for version 5 and work is already in progress, but I cannot tell how long it will take until it is ready for release.

The longer answer is that it is not as easy to implement for the general case. The reason is that once the addressbooks have been added, rcmcarddav has no linking in its data model that these addressbooks belong to one account. This change to the data model will happen in v5, and pave the road for performing a rediscovery. However, presets are a special case, because for these we do in fact have a link in that we know which addressbooks belong to which preset. I have pushed the feature to the master branch if you want to try. Rediscovery of presets will take place upon login to roundcube.

**Comment by hpkunz** [comment]
hey, thanks, very much. this is exactly what I needed. On first tests everything works as expected.

thanks for your explanations above, I understand why this is not yet working for the general case (user added address books). I (wrongly) assumed that address books that are added in one go (using discovery) are grouped together in a namespace, much like the presets.

Best,
Hp
